### PR TITLE
Testing Plan Phase 2

### PR DESCRIPTION
**Pull request by BowTiedRadone**
This PR implements **Phase 2** of a broader testing strategy, adding property-based tests to expand input coverage beyond unit tests. Focused on `smart-wallet-standard` and `csw-registry` using `fast-check` for randomized testing scenarios.

**Key Changes**

- **Property Testing**: Comprehensive property-based tests for both `smart-wallet-standard` and `csw-registry` covering the main functionalities: transfers, registrations, state management, etc.
- **Input Coverage**: Tests protocol behavior across randomized inputs, providing immediate feedback on changes and increasing the chance of catching edge cases
- **Architecture**: Updates `smart-wallet-standard` to reference simnet deployer address `extension-trait` for improved testability and preparation for **Phase 3**
- **Invariant Validation**: Verifies balance conservation, authorization, ID sequencing, and state consistency properties across various inputs
- **CI/CD**: New property-based tests are automatically picked up by the `tests` workflow, `tests` workflow was updated to only push `lcov.info`
- **Cleanup**: Removes the verbose `coverage-html` directory, updates tests to use the `clarigen` generated interface for project contracts error codes

**Usage**
```bash
pnpm install
pnpm generate  
pnpm test
```

**Notes**

This phase builds on **[Phase 1](https://github.com/polimartlabs/csw-locker-contracts/pull/2)**'s unit test foundation by exploring broader input spaces through randomized testing, ensuring system robustness across diverse scenarios.

### CODE REVIEW

File: contracts/smart-wallet-standard.clar [hunk 1]
```diff
@@ -1,7 +1,9 @@
 ;; title: smart-wallet-standard
 ;; version: 1
 ;; summary: Extendible single-owner smart wallet with standard SIP-010 and SIP-009 support
-(use-trait extension-trait .extension-trait.extension-trait)
+
+;; Using deployer address for testing.
+(use-trait extension-trait 'ST1PQHQKV0RJXZFY1DGX8MNSNYVE3VGZJSRTPGZGM.extension-trait.extension-trait)
```
[the remark this answers is not in the document]
Comment (BowTiedRadone) [reply contracts/smart-wallet-standard.clar]: 59a7371
